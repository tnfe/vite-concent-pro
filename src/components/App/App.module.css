.header {
  z-index: 88;
  position: relative;
}

.headerSetting {
  position: absolute;
  right: 22px;
  top: 23px;
  font-size: 19px;
}


.userIconWrap {
  position: absolute;
  min-width: 100px;
  right: 52px;
  font-size: 15px;
}


.headerSetting:hover {
  cursor: pointer;
}

.siderWrap {
  position: fixed;
  padding-top: 64px;
  height: 100vh;
  overflow-y: auto;
  overflow-x: hidden;
  top: 0;
}

.contentWrap {
  position: relative;
  background: #fff;
  /* background: blue; */
  padding: 24px;
  margin: 0;
  min-height: calc(100vh - 152px);
}
