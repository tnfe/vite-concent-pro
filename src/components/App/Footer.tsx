import React from 'react';

const Footer = () => (
  <div style={{textAlign: 'center', padding: '5px'}}>
    Powered by <a href="https://github.com/concentjs/concent" target="blank">Concent 2021</a>
  </div>
);

export default Footer;
