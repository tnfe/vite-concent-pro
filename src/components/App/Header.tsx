import React from 'react';
import { Layout, Popover, Switch, Tag, Avatar } from 'antd';
import { SwitchChangeEventHandler } from 'antd/lib/switch/index';
import { useConcent, SettingsType } from 'concent';
import { SketchPicker, ColorChangeHandler, ColorResult } from 'react-color'
import styles from './App.module.css';
import { SettingOutlined } from '@ant-design/icons';
import { CtxDe } from 'types/store';
import { Blank, VerticalBlank } from 'components/dumb/general';

interface ISettingPanelProps {
  color: string;
  onChangeComplete: ColorChangeHandler;
  onWebsiteColorChange: ColorChangeHandler;
  siderThemeChecked: boolean;
  onSiderThemeChange: SwitchChangeEventHandler;
  headerThemeChecked: boolean;
  onHeaderThemeChange: SwitchChangeEventHandler;
}

function SettingPanel({
  color, onChangeComplete, onWebsiteColorChange, siderThemeChecked, onSiderThemeChange, headerThemeChecked, onHeaderThemeChange,
}: ISettingPanelProps) {
  return (
    <div style={{ padding: '12px 28px' }}>
      <Tag color="geekblue">站点主题设置：</Tag>
      <VerticalBlank />
      <SketchPicker color={color} onChange={onWebsiteColorChange} onChangeComplete={onChangeComplete} />
      <VerticalBlank />
      <div>
        <Tag color="geekblue">边栏设置：</Tag>
        <Blank />
        <Switch checkedChildren="关闭暗黑边栏" unCheckedChildren="开启暗黑边栏" checked={siderThemeChecked} onChange={onSiderThemeChange} />
      </div>
      <VerticalBlank />
      <div>
        <Tag color="geekblue">顶栏设置：</Tag>
        <Blank />
        <Switch checkedChildren="关闭暗黑顶栏" unCheckedChildren="开启暗黑顶栏" checked={headerThemeChecked} onChange={onHeaderThemeChange} />
      </div>
    </div>
  );
}

type Se = SettingsType<typeof setup>;
function setup(ctx: CtxDe) {
  const { globalReducer: grd } = ctx;
  return {
    onWebsiteColorChange(colorResult: ColorResult) {
      grd.changeThemeColor(colorResult.hex);
    },
    onSiderThemeChange(checked: boolean) {
      grd.switchSiderTheme(checked);
    },
    onHeaderThemeChange(checked: boolean) {
      grd.switchHeaderTheme(checked);
    },
  }
}

function AppHeader() {
  const {
    globalState: st, globalComputed: gcu, settings: se
  } = useConcent<{}, CtxDe<{}, Se>>({ setup, tag: 'Header' });
  const uiContent = <SettingPanel color={st.themeColor} onWebsiteColorChange={se.onWebsiteColorChange}
    onChangeComplete={se.onWebsiteColorChange} siderThemeChecked={gcu.siderThemeSwitchChecked}
    headerThemeChecked={gcu.headerThemeSwitchChecked}
    onSiderThemeChange={se.onSiderThemeChange} onHeaderThemeChange={se.onHeaderThemeChange} />;

  return (
    <Layout.Header className={styles.header} style={gcu.headerStyle}>
      <div className={styles.userIconWrap}>
        <Avatar size={40} src={st.uesrIcon} />
        <Blank width="8px" />
        {st.userName}
      </div>
      <Popover placement="bottomLeft" content={uiContent} title="主题设置">
        <SettingOutlined className={styles.headerSetting} style={{ color: gcu.headerStyle.color }} />
      </Popover>
    </Layout.Header>
  );
}

export default React.memo(AppHeader);
