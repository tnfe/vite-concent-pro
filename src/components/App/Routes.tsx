/**
 * 根据 configs/menus 配置组装整个应用路由系统
 */
import React, { Suspense } from 'react';
import { Route, Switch } from 'react-router-dom';
import { cst, register } from 'concent';
import { getUrlChangedEvName } from 'react-router-concent';
import { flatedMenus } from 'configs/derived/menus';
import { IMenuItem } from 'configs/menus';
import NotFound from 'pages/NotFound';
import { CtxDe } from 'types/store';

class Routes extends React.Component {
  ctx = {} as CtxDe;
  errOccurred = false;

  state = {
    err: '',
  };

  $$setup() {
    this.ctx.on(getUrlChangedEvName(), () => {
      if (this.errOccurred) {
        this.errOccurred = false;
        this.setState({err: ''});
      }
    });
  }

  componentDidCatch(err: any) {
    this.errOccurred = true;
    this.setState({err: err.message});
  }

  render() {
    if (this.errOccurred) {
      return <h1 style={{color: 'red'}}>
        当前路由页面崩溃，请联系 xxx开发者 做进一步跟踪，如果是开发者，可打开console查看具体错误,
        如想继续访问当前页面，可刷新留浏览器重试。
      </h1>;
    }

    let homeMenuItem = null as IMenuItem | null;
    const uiRoutes = flatedMenus.map((item) => {
      const {path, Component, exact = true} = item;
      if (item.isHomePage) homeMenuItem = item;
      return <Route key={path} exact={exact} path={path} component={Component}/>;
    });

    let uiHomeRoute = '' as React.ReactNode;
    if (homeMenuItem) {
      uiHomeRoute = <Route key={'/'} exact={true} path={'/'} component={homeMenuItem.Component}/>
    }

    return (
      <Suspense fallback={<div>Loading...</div>}>
        <Switch>
          {uiRoutes}
          {uiHomeRoute}
          <Route component={NotFound}/>
        </Switch>
      </Suspense>
    );
  }
}

export default register(cst.MODULE_DEFAULT)(Routes);
