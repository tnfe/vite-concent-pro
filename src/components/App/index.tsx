import 'configs/runConcent';
// +++ node modules +++
import React from 'react';
import { ConnectRouter, getUrlChangedEvName } from 'react-router-concent';
import { BrowserRouter } from 'react-router-dom';
import { Layout, Spin } from 'antd';
import { useConcent } from 'concent';
// +++ project modules +++
import { CtxDe } from 'types/store';
import SiderSwitchIcon from 'components/biz-dumb/SiderSwitchIcon';
import { getBasename, getRelativeRootPath } from 'services/appPath';
import { path2menuGroup, path2menuItem } from 'configs/derived/menus';
// +++ local modules +++
import styles from './App.module.css';
import Routes from './Routes';
import Sider from './Sider';
import Footer from './Footer';
import Header from './Header';

const {Content} = Layout;

function setup({effect, on, globalReducer}: CtxDe) {

  const setAppPathLabel = () => {
    const curAppPath = getRelativeRootPath();
    const menuItem = path2menuItem[curAppPath];
    if (menuItem) {
      const label2 = menuItem.label;
      const menuGroup = path2menuGroup[curAppPath];
      let label1 = '';
      if (menuGroup) {
        label1 = `${menuGroup.label}/`;
      }
      const dom = document.querySelector('#appPathLabel') as any;
      if (!dom) return;
      dom.innerText = `${label1}${label2}`;
    }
  };

  effect(() => {
    setAppPathLabel();
    globalReducer.prepareApp()
  }, []);

  on(getUrlChangedEvName(), (param, action, history) => {
    console.log(param, action, history);
    setAppPathLabel();
  });
}

function App() {
  const {globalReducer, globalState, globalComputed} = useConcent<{}, CtxDe>({setup, tag: 'App'});
  const siderVisible = globalState.siderVisible;

  let uiContentArea = '' as React.ReactNode;
  if (!globalState.isAppReady) {
    uiContentArea = <Spin>系统初始化中...</Spin>
  } else {
    uiContentArea = <Routes/>;
  }

  return (
    <Layout>
      <Layout>
        <Header/>
      </Layout>
      <Layout>
        <SiderSwitchIcon des={globalComputed.siderIconDes} onClick={globalReducer.toggleSiderVisible}/>
        {siderVisible && <Sider/>}
      </Layout>
      <Layout style={globalComputed.contentLayoutStyle}>
        <h5 id="appPathLabel"></h5>
        <Content id="appContentArea" className={styles.contentWrap}>
          {uiContentArea}
        </Content>
        <Footer/>
      </Layout>
    </Layout>
  );
}

export default React.memo(() => {
  return (
    <BrowserRouter basename={`/${getBasename()}`}>
      <ConnectRouter callUrlChangedOnInit={true}>
        <App/>
      </ConnectRouter>
    </BrowserRouter>
  );
});
