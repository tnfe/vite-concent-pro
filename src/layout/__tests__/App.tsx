import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../index';

// 组件实例
let ins = null as any;

/**
 * @author fancyzhong
 * @priority P0
 * @casetype unit
 */
describe('App Render', () => {
  beforeAll(() => {
    // ins = mount(<App />);
    // avoid PrettyFormatPluginError: Invalid string lengthRangeError: Invalid string length
    ins = render(<App />);
  });

  test('generate App snapshot', () => {
    // expect(ins).toMatchSnapshot();
    // avoid PrettyFormatPluginError: Invalid string lengthRangeError: Invalid string length
    expect(toJson(ins)).toMatchSnapshot();
  });
});
