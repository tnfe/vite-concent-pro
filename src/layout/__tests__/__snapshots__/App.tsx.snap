// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render generate App snapshot 1`] = `
<section
  class="ant-layout"
>
  <header
    class="ant-layout-header header"
    style="margin-left:48px;background-color:white;color:var(--theme-color);z-index:4;display:none"
  >
    <div
      class="headerRightAreaWrap"
      style="color:var(--theme-color)"
    >
      <span
        class="ant-avatar ant-avatar-circle userIconWarp"
        style="width:32px;height:32px;line-height:32px;font-size:18px"
      >
        <img
          src=""
        />
      </span>
      <div
        style="display:inline-block;width:8px;height:16px"
      />
      <div
        style="display:inline-block;width:8px;height:16px"
      />
    </div>
  </header>
  <div
    class="quickNavBarWrapBase quickNavBarLightWrap smallScBar"
    style="box-sizing:border-box;padding-left:48px;z-index:5;position:fixed;left:0;right:0;top:0"
  >
    <div
      class="ant-tabs ant-tabs-top ant-tabs-editable ant-tabs-card ant-tabs-editable-card"
      style="padding-left:3px;display:inline-block"
    >
      <div
        class="ant-tabs-nav"
        role="tablist"
      >
        <div
          class="ant-tabs-nav-wrap"
        >
          <div
            class="ant-tabs-nav-list"
            style="transform:translate(0px, 0px)"
          >
            <div
              class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
            />
          </div>
        </div>
        <div
          class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
        >
          <button
            aria-controls="null-more-popup"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-hidden="true"
            class="ant-tabs-nav-more"
            id="null-more"
            style="visibility:hidden;order:1"
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="ant-tabs-content-holder"
      >
        <div
          class="ant-tabs-content ant-tabs-content-top"
        />
      </div>
    </div>
    <div
      class="headerSettingWrapInBar"
    >
      <button
        aria-checked="false"
        class="ant-switch"
        role="switch"
        style="vertical-align:text-bottom"
        type="button"
      >
        <div
          class="ant-switch-handle"
        />
        <span
          class="ant-switch-inner"
        >
          浅
        </span>
      </button>
      <div
        style="display:inline-block;width:6px;height:16px"
      />
      <span
        aria-label="setting"
        class="anticon anticon-setting  gHover"
        role="img"
        style="color:var(--theme-color)"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          data-icon="setting"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
          />
        </svg>
      </span>
    </div>
  </div>
  <span
    style="display:none"
  />
  <div
    class="ccpro-drawer__icon"
    style="position:fixed;top:48%;z-index:999999"
  >
    <span>
      <span
        aria-label="right-circle"
        class="anticon anticon-right-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M666.7 505.5l-246-178A8 8 0 00408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
          />
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    class="siderWrap"
    style="background-color:#001529;width:48px"
  >
    <div
      class="logoWrap"
      style="background-image:url(https://raw.githubusercontent.com/fantasticsoul/assets/master/c2pro/short_logo.png);width:48px"
    />
    <div
      class="siderMenusWrap smallScBar"
      style="width:48px"
    >
      <ul
        class="ant-menu ant-menu-dark ant-menu-inline-collapsed layout-sider ant-menu-root ant-menu-vertical"
        role="menu"
        style="height:100%;border-right:0"
      >
        <li
          class="ant-menu-item ant-menu-item-selected"
          role="menuitem"
        >
          <span
            aria-label="form"
            class="anticon anticon-form ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="form"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M904 512h-56c-4.4 0-8 3.6-8 8v320H184V184h320c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V520c0-4.4-3.6-8-8-8z"
              />
              <path
                d="M355.9 534.9L354 653.8c-.1 8.9 7.1 16.2 16 16.2h.4l118-2.9c2-.1 4-.9 5.4-2.3l415.9-415c3.1-3.1 3.1-8.2 0-11.3L785.4 114.3c-1.6-1.6-3.6-2.3-5.7-2.3s-4.1.8-5.7 2.3l-415.8 415a8.3 8.3 0 00-2.3 5.6zm63.5 23.6L779.7 199l45.2 45.1-360.5 359.7-45.7 1.1.7-46.4z"
              />
            </svg>
          </span>
          <span>
            <a
              href="/demo"
            >
              首页
            </a>
          </span>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-vertical"
          role="menuitem"
        >
          <div
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="button"
          >
            <span
              aria-label="ordered-list"
              class="anticon anticon-ordered-list"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ordered-list"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M920 760H336c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0-568H336c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H336c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM216 712H100c-2.2 0-4 1.8-4 4v34c0 2.2 1.8 4 4 4h72.4v20.5h-35.7c-2.2 0-4 1.8-4 4v34c0 2.2 1.8 4 4 4h35.7V838H100c-2.2 0-4 1.8-4 4v34c0 2.2 1.8 4 4 4h116c2.2 0 4-1.8 4-4V716c0-2.2-1.8-4-4-4zM100 188h38v120c0 2.2 1.8 4 4 4h40c2.2 0 4-1.8 4-4V152c0-4.4-3.6-8-8-8h-78c-2.2 0-4 1.8-4 4v36c0 2.2 1.8 4 4 4zm116 240H100c-2.2 0-4 1.8-4 4v36c0 2.2 1.8 4 4 4h68.4l-70.3 77.7a8.3 8.3 0 00-2.1 5.4V592c0 2.2 1.8 4 4 4h116c2.2 0 4-1.8 4-4v-36c0-2.2-1.8-4-4-4h-68.4l70.3-77.7a8.3 8.3 0 002.1-5.4V432c0-2.2-1.8-4-4-4z"
                />
              </svg>
            </span>
            <span>
              list示例集合
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
        >
          <span
            aria-label="question"
            class="anticon anticon-question ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="question"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M764 280.9c-14-30.6-33.9-58.1-59.3-81.6C653.1 151.4 584.6 125 512 125s-141.1 26.4-192.7 74.2c-25.4 23.6-45.3 51-59.3 81.7-14.6 32-22 65.9-22 100.9v27c0 6.2 5 11.2 11.2 11.2h54c6.2 0 11.2-5 11.2-11.2v-27c0-99.5 88.6-180.4 197.6-180.4s197.6 80.9 197.6 180.4c0 40.8-14.5 79.2-42 111.2-27.2 31.7-65.6 54.4-108.1 64-24.3 5.5-46.2 19.2-61.7 38.8a110.85 110.85 0 00-23.9 68.6v31.4c0 6.2 5 11.2 11.2 11.2h54c6.2 0 11.2-5 11.2-11.2v-31.4c0-15.7 10.9-29.5 26-32.9 58.4-13.2 111.4-44.7 149.3-88.7 19.1-22.3 34-47.1 44.3-74 10.7-27.9 16.1-57.2 16.1-87 0-35-7.4-69-22-100.9zM512 787c-30.9 0-56 25.1-56 56s25.1 56 56 56 56-25.1 56-56-25.1-56-56-56z"
              />
            </svg>
          </span>
          <span>
            <a
              href="/template"
            >
              template
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
        >
          <span
            aria-label="plus-circle"
            class="anticon anticon-plus-circle ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="plus-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
              />
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
            </svg>
          </span>
          <span>
            <a
              href="/counter"
            >
              计数器
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
        >
          <span
            aria-label="plus-circle"
            class="anticon anticon-plus-circle ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="plus-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
              />
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
            </svg>
          </span>
          <span>
            <a
              href="/counter2"
            >
              计数器2
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
        >
          <span
            aria-label="question"
            class="anticon anticon-question ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="question"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M764 280.9c-14-30.6-33.9-58.1-59.3-81.6C653.1 151.4 584.6 125 512 125s-141.1 26.4-192.7 74.2c-25.4 23.6-45.3 51-59.3 81.7-14.6 32-22 65.9-22 100.9v27c0 6.2 5 11.2 11.2 11.2h54c6.2 0 11.2-5 11.2-11.2v-27c0-99.5 88.6-180.4 197.6-180.4s197.6 80.9 197.6 180.4c0 40.8-14.5 79.2-42 111.2-27.2 31.7-65.6 54.4-108.1 64-24.3 5.5-46.2 19.2-61.7 38.8a110.85 110.85 0 00-23.9 68.6v31.4c0 6.2 5 11.2 11.2 11.2h54c6.2 0 11.2-5 11.2-11.2v-31.4c0-15.7 10.9-29.5 26-32.9 58.4-13.2 111.4-44.7 149.3-88.7 19.1-22.3 34-47.1 44.3-74 10.7-27.9 16.1-57.2 16.1-87 0-35-7.4-69-22-100.9zM512 787c-30.9 0-56 25.1-56 56s25.1 56 56 56 56-25.1 56-56-25.1-56-56-56z"
              />
            </svg>
          </span>
          <span>
            <a
              href="/step-form"
            >
              分步表单
            </a>
          </span>
        </li>
        <li
          class="ant-menu-item"
          role="menuitem"
        >
          <span
            aria-label="right-circle"
            class="anticon anticon-right-circle ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="right-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M666.7 505.5l-246-178A8 8 0 00408 334v46.9c0 10.2 4.9 19.9 13.2 25.9L566.6 512 421.2 617.2c-8.3 6-13.2 15.6-13.2 25.9V690c0 6.5 7.4 10.3 12.7 6.5l246-178c4.4-3.2 4.4-9.8 0-13z"
              />
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
            </svg>
          </span>
          <span>
            <a
              href="/demo-use-setup"
            >
              useSetup
            </a>
          </span>
        </li>
      </ul>
    </div>
    <button
      class="ant-btn ant-btn-primary"
      style="width:100%;border:none"
      type="button"
    >
      <span
        aria-label="double-right"
        class="anticon anticon-double-right"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="double-right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 00188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 00492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
          />
        </svg>
      </span>
    </button>
  </div>
  <section
    class="ant-layout"
    style="margin-left:48px;padding-top:32px;min-height:calc(100vh - 45px);overflow-x:auto"
  >
    <div
      style="padding:64px"
    >
      <div
        class="ant-skeleton ant-skeleton-with-avatar"
      >
        <div
          class="ant-skeleton-header"
        >
          <span
            class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
          />
        </div>
        <div
          class="ant-skeleton-content"
        >
          <h3
            class="ant-skeleton-title"
            style="width:50%"
          />
          <ul
            class="ant-skeleton-paragraph"
          >
            <li />
            <li />
            <li />
            <li />
          </ul>
        </div>
      </div>
      <div
        class="ant-skeleton ant-skeleton-with-avatar"
      >
        <div
          class="ant-skeleton-header"
        >
          <span
            class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
          />
        </div>
        <div
          class="ant-skeleton-content"
        >
          <h3
            class="ant-skeleton-title"
            style="width:50%"
          />
          <ul
            class="ant-skeleton-paragraph"
          >
            <li />
            <li />
            <li />
            <li />
          </ul>
        </div>
      </div>
      <div
        class="ant-spin-nested-loading"
      >
        <div>
          <div
            class="ant-spin ant-spin-spinning"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
          </div>
        </div>
        <div
          class="ant-spin-container ant-spin-blur"
        >
          <div
            style="text-align:center"
          >
            系统初始化中...
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="footerWrap"
  >
    © 2021 
    <a
      href="https://github.com/tnfe/concent-pro"
      target="blank"
    >
      Concent Pro
    </a>
    , powered by 
    <a
      href="https://github.com/concentjs/concent"
      target="blank"
    >
      Concent 
    </a>
    <a
      href="https://github.com/concentjs/concent"
      target="blank"
    >
      <span
        aria-label="github"
        class="anticon anticon-github"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="github"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
          />
        </svg>
      </span>
    </a>
  </div>
</section>
`;
